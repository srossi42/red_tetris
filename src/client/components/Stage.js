import React from 'react';

import Cell from './Cell';

const Stage = ({ stage }) => {

    const TETROMINOS = {
        0: { color: "0, 0, 0" },
        I: { color: "80, 227, 230" },
        J: { color: "36, 95, 223" },
        L: { color: "223, 173, 36" },
        O: { color: "223, 217, 36" },
        S: { color: "48, 211, 56" },
        T: { color: "132, 61, 198" },
        Z: { color: "227, 78, 78" }
      };

    return stage.map(row => row.map((cell, x) =>
        <Cell key={x} type={cell[0]} color={TETROMINOS[cell[0]].color}/>
    ))
}


export const StagePreview = ({ stage }) => {

    const TETROMINOS = {
        0: { color: "255, 255, 255" },
        I: { color: "80, 227, 230" },
        J: { color: "36, 95, 223" },
        L: { color: "223, 173, 36" },
        O: { color: "223, 217, 36" },
        S: { color: "48, 211, 56" },
        T: { color: "132, 61, 198" },
        Z: { color: "227, 78, 78" }
      };

    return stage.map(row => row.map((cell, x) => 
        <Cell key={x} type={cell[0]} color={TETROMINOS[cell[0]].color}/>
    ))
}



export default Stage