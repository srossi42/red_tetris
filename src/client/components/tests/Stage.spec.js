import React from 'react'
import { shallow } from 'enzyme';
import Stage, { StagePreview } from '../stage'
import { createStage, createStagePreview } from '../../gameHelper'
import toJson from "enzyme-to-json"

describe('<Stage/> Component', () => {
  it('rendering correctly Stage', () => {
    let stage = createStage()
    let wrapper = shallow(
      <Stage stage={stage}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('rendering correctly StagePreview', () => {
    let stage = createStagePreview()
    let wrapper = shallow(
      <StagePreview stage={stage}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
