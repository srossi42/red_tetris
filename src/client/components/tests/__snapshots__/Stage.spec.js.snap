// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage/> Component rendering correctly Stage 1`] = `
Array [
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="0, 0, 0"
    key="9"
    type={0}
  />,
]
`;

exports[`<Stage/> Component rendering correctly StagePreview 1`] = `
Array [
  <Memo(Cell)
    color="255, 255, 255"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="9"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="0"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="1"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="2"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="3"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="4"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="5"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="6"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="7"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="8"
    type={0}
  />,
  <Memo(Cell)
    color="255, 255, 255"
    key="9"
    type={0}
  />,
]
`;
